<h1>Activity #1 - Palmer Penguins Data Analytics</h1>
<hr>
<h3>Analyst: Carl Kien Carabido</h3>

In [1]:
#import Libraries
import numpy as np
import pandas as pd

<h3>Converting Dataset(s) into DataFrame</h3>
<h3>Requirements:</h3>
<ul>
    <li>Name of the DataFrame</li>
    <li>File Path</li>
</ul>

In [2]:
#import csv file
penguins = pd.read_csv('datasets\\penguins.csv')
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [3]:
# penguins.describe()
#.T for transpose or to flip
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Checking for Null Values</h3>

In [4]:
#pengruins.isnull()
#sum() for summarize
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h3>Filling up the Missing Values</h3>

In [5]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [6]:
penguins['bill_length_mm'].mean()
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [8]:
penguins['bill_depth_mm'].mean()
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [9]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [10]:
penguins['flipper_length_mm'].mean()
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [11]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [12]:
penguins['body_mass_g'].mean()
penguins['body_mass_g'] = penguins['flipper_length_mm'].fillna(penguins['body_mass_g'].mean())

In [13]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
dtype: int64

In [14]:
penguins['sex'].mode()
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [15]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h3>Changing the Data Type</h3>

In [16]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [17]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['year'] = penguins['year'].astype('category')

<h3>Making a column</h3>

In [18]:
penguins.columns

Index(['species', 'island', 'bill_length_mm', 'bill_depth_mm',
       'flipper_length_mm', 'body_mass_g', 'sex', 'year'],
      dtype='object')

In [19]:
#it assumes you will make a new column
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000, 2)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0     181.000000    male  2007     0.18  
1     186.000000  female  2007     0.19  
2     195.000000  female  2007     0.20  
3     200.915205    male  2007     0.20  
4     193.000000  female  2007     0.19  
..           ...     ...   ...      ...  
339   207.000000    male  2009     0.21  
340   202.000000  female  2009     0.20  
341   193.000000    male  2009     0.19  
342   210.000000    male  2009     0.21  
343   198.000000  female  2009     0.20  

[344 rows x 9 columns]

<h3>Creating individual DataFrames for each species.</h3>

In [20]:
species1 = penguins[penguins['species'] == 'Chinstrap']
species2 = penguins[penguins['species'] == 'Adelie']
species3 = penguins[penguins['species'] == 'Gentoo']
# species1
# species2
# species3

<h2>Explanatory Data Analytics</h2>
<hr>
<h3>Question #1</h3>
<h4>What species has the longest Bill Length?</h4>

In [21]:
penguins['bill_length_mm'].max()

np.float64(59.6)

In [22]:
penguins[penguins['bill_length_mm'] == 59.6]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
185  Gentoo  Biscoe            59.6           17.0              230.0   

     body_mass_g   sex  year  mass_kg  
185        230.0  male  2007     0.23

<h3>Insight:</h3>
<p>The species that has the longest Bill is Gentoo.</p>
<hr>

<h3>Question #2</h3>
<h4>What is the species of the heaviest penguin?</h4>

In [23]:
penguins['mass_kg'].max()

np.float64(0.23)

In [24]:
penguins[penguins['mass_kg'] == 0.23]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
153  Gentoo  Biscoe            50.0           16.3              230.0   
185  Gentoo  Biscoe            59.6           17.0              230.0   
215  Gentoo  Biscoe            54.3           15.7              231.0   
217  Gentoo  Biscoe            49.8           16.8              230.0   
219  Gentoo  Biscoe            49.5           16.2              229.0   
227  Gentoo  Biscoe            48.6           16.0              230.0   
237  Gentoo  Biscoe            50.8           17.3              228.0   
241  Gentoo  Biscoe            52.1           17.0              230.0   
243  Gentoo  Biscoe            52.2           17.1              228.0   
247  Gentoo  Biscoe            50.8           15.7              226.0   
253  Gentoo  Biscoe            55.9           17.0              228.0   
255  Gentoo  Biscoe            49.1           15.0              228.0   
263  Gentoo  Biscoe            49.8           15.9              229.0   
265  Gentoo  Biscoe            51.5           16.3              230.0   
267  Gentoo  Biscoe            55.1           16.0              230.0   

     body_mass_g   sex  year  mass_kg  
153        230.0  male  2007     0.23  
185        230.0  male  2007     0.23  
215        231.0  male  2008     0.23  
217        230.0  male  2008     0.23  
219        229.0  male  2008     0.23  
227        230.0  male  2008     0.23  
237        228.0  male  2009     0.23  
241        230.0  male  2009     0.23  
243        228.0  male  2009     0.23  
247        226.0  male  2009     0.23  
253        228.0  male  2009     0.23  
255        228.0  male  2009     0.23  
263        229.0  male  2009     0.23  
265        230.0  male  2009     0.23  
267        230.0  male  2009     0.23

<h3>Insight:</h3>
<p>The Heaviest species is Gentoo with the weigth of 0.23 kg</p>
<hr>

<h3>Question #3</h3>
<h4>In what year was the longest flipper length in a penguin recorded?</h4>

In [25]:
penguins['flipper_length_mm'].max()

np.float64(231.0)

In [26]:
penguins[penguins['flipper_length_mm'] == 231.0]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
215  Gentoo  Biscoe            54.3           15.7              231.0   

     body_mass_g   sex  year  mass_kg  
215        231.0  male  2008     0.23

<h3>Insight:</h3>
<p>The penguin was recorded in 2008; it was a male Gentoo penguin.</p>
<hr>

<h3>Question #4</h3>
<h4>How many of each species are there on each island?</h4>

In [35]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [34]:
gento_biscoe = species3[species3['island'] == 'Biscoe']
adelie_biscoe = species2[species2['island'] == 'Biscoe']
chinstrap_biscoe = species1[species1['island'] == 'Biscoe']
chinstrap_biscoe.count() # Shows 0 entries
adelie_biscoe.count() # Shows 44 entries
gento_biscoe.count() # Shows 124 entries

species              124
island               124
bill_length_mm       124
bill_depth_mm        124
flipper_length_mm    124
body_mass_g          124
sex                  124
year                 124
mass_kg              124
dtype: int64

In [42]:
chinstrap_biscoe = species1[species1['island'] == 'Dream']
adelie_biscoe = species2[species2['island'] == 'Dream']
gento_biscoe = species3[species3['island'] == 'Dream']
chinstrap_biscoe.count() # Shows 68 entries
adelie_biscoe.count() # Shows 56 entries
gento_biscoe.count() # Shows 0 entries

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
mass_kg              0
dtype: int64

In [41]:
chinstrap_biscoe = species1[species1['island'] == 'Torgersen']
adelie_biscoe = species2[species2['island'] == 'Torgersen']
gento_biscoe = species3[species3['island'] == 'Torgersen']
chinstrap_biscoe.count() # Shows 0 entries
adelie_biscoe.count() # Shows 52 entries
gento_biscoe.count() # Shows 0 entries

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
mass_kg              0
dtype: int64

<h3>Insight:</h3>
<p>Adelie penguins are found on all three islands. Biscoe has mostly gentoo and some adelie, but no chinstrap. Dream has chinstrap and adelie, but no gentoo. Torgersen only has adelie. It looks like each species prefers certain islands, except for adelies who are found everywhere.</p>
<hr>

<h3>Question #5</h3>
<h4>Which island has the lightest penguin?</h4>

In [48]:
penguins[penguins["body_mass_g"] == penguins["body_mass_g"].min()]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
28  Adelie  Biscoe            37.9           18.6              172.0   

    body_mass_g     sex  year  mass_kg  
28        172.0  female  2007     0.17

<h3>Insight:</h3>
<p>The lightest penguin recorded in the dataset is an Adelie female found on Biscoe Island in 2007, weighing only 172 grams.</p>
<hr>

<h3>Question #6</h3>
<h4>What species has the widest bill (on average)?</h4>

In [50]:
print("Adelie:", species2["bill_depth_mm"].mean())
print("Chinstrap:", species1["bill_depth_mm"].mean())
print("Gentoo:", species3["bill_depth_mm"].mean())

Adelie: 18.338494536780548
Chinstrap: 18.42058823529412
Gentoo: 14.999606206376153


<h3>Insight:</h3>
<p>Chinstrap penguins have the widest bills on average.</p>
<hr>

<h3>Question #7</h3>
<h4>Which year has the most penguin records?</h4>

In [51]:
penguins["year"].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3>Insight:</h3>
<p>2009 has the most penguin records.</p>
<hr>